android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp

```diff
@@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
 {
   return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
 }
+
+JNIEXPORT jstring JNICALL
+Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
+     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
+{
+  Bookmark const * bmk = getBookmark(catId, bmkId);
+  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
```

Comment:
Дима, адрес и не должен задаваться ни в каких конструкторах.

Вычисление адреса - трудоемкая операция, ее нужно точечно делать только тогда, когда нужно.

Но мне не понятно, откуда тогда вызывается этот код. Если из pp, то адрес там должен браться из самого pp, уже вычисленный. А если это для шаринга, то при чем тут букмарка?

Посмотри на реализацию. Для вычисления адреса нужна либо точка, либо фича. Это никакого отношения не имеет к букмаркам и не понятно почему вообще тут находится.

> On 25.3.2016, at 19.00, Dmitry Yunitsky <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp:
> 
> > @@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
> >  {
> >    return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
> >  }
> > +
> > +JNIEXPORT jstring JNICALL
> > +Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
> > +     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
> > +{
> > +  Bookmark const * bmk = getBookmark(catId, bmkId);
> > +  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
> Ещё раз - адрес букмарок ВСЕГДА берётся этим методом. В конструкторе букмарки мы НЕ задаём адрес, чтобы он был легковесней. Потому что при конструировании множества букмарок при просмотре списка - нужно получать их быстро. А адрес там НЕ нужен. По поводу неудачности - надо дорефакторить в ядре букмарки, тогда и здесь можно сделать красиво. Пропадали названия как раз потому, что я ошибочно посчитал, что названия букмарок уже можно брать из place_page::Info.
> …
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
>